src/server/server_family.cc

```diff
@@ -871,7 +871,7 @@ error_code ServerFamily::DoSave(bool new_version, Transaction* trans, string* er
       const auto scripts = script_mgr_->GetLuaScripts();
       auto& summary_snapshot = snapshots[shard_set->size()];
       summary_snapshot.reset(new RdbSnapshot(fq_threadpool_.get()));
-      if (ec = DoPartialSave(filename, path, start, scripts, summary_snapshot.get(), nullptr)) {
+      if ((ec = DoPartialSave(filename, path, start, scripts, summary_snapshot.get(), nullptr))) {
```

Comment:
Same as the previous comment - I think that it would look more like the intent is to assign then check. assign with side effect is the classic that you;re not sure if this is a bug or you're using the side effect